apps/prairielearn/src/ee/pages/instructorAiGenerateDrafts/instructorAiGenerateDrafts.ts

```diff
@@ -29,4 +33,42 @@ router.get(
   }),
 );
 
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_permission_edit) {
+      throw new error.HttpStatusError(403, 'Access denied (must be course editor)');
+    }
+
+    if (req.body.__action === 'delete_drafts') {
+      const questions = await queryRows(
+        sql.select_draft_questions_by_course_id,
+        { course_id: res.locals.course.id.toString() },
+        z.object({ id: IdSchema, qid: z.string() }),
+      );
+
+      const client = getCourseFilesClient();
+
+      const result = await client.batchDeleteQuestions.mutate({
+        course_id: res.locals.course.id,
+        user_id: res.locals.user.user_id,
+        authn_user_id: res.locals.authn_user.user_id,
+        has_course_permission_edit: res.locals.authz_data.has_course_permission_edit,
+        question_ids: questions.map((question) => question.id),
+      });
+
+      if (result.status === 'error') {
+        throw new error.HttpStatusError(
+          500,
+          `Cannot delete all draft questions in: (${questions.map((question) => question.qid).join(', ')})`,
```

Comment:
No need to list QIDs, they'll still be visible if the user navigates back (and I doubt they'll care anyways):

```suggestion
          'Failed to delete all draft questions.',
```

After changing this, you can update the `select_draft_questions_by_course_id` query to no longer return QIDs.

Reply:
Done in 16aa671758e5d243976b2f58594e0032923eb6f8.